### PR TITLE
[CodeCamp2023-338] New Version of config Adapting Swin Transformer Algorithm

### PR DESCRIPTION
**Pull request by timerring**
Adapting New Version of config Adapting Swin Transformer Algorithm, based on https://github.com/open-mmlab/OpenMMLabCamp/discussions/338

## Motivation

MMEngine has introduced a novel configuration mechanism, utilizing which we will perform configuration file adaptation.

## Modification

Add 3 config python files in `mmpretrain/configs/_base_/datasets`
Add 12 config python files in `mmpretrain/configs/_base_/models`
Add 1 config python file in `mmpretrain/configs/_base_/schedules`
Add 7 config python files in `mmpretrain/configs/swin_transformer`
Add 12 config python files in `mmpretrain/configs/swin_transformer_v2`

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by timerring** [comment]
Can you please help me with the code review? thanks a lot. @fangyixiao18

**Comment by fangyixiao18** [comment]
please remove the folder `_base_/models/swin_transformer` and `_base_/models/swin_transformer_v2`, but only keep one file to set model config like `swin_transformer_base.py` and `swin_transformer_v2_base.py`,and the training configs all inherit from this `swin_transformer_base.py` file

File: mmpretrain/configs/swin_transformer/swin-base_16xb64_in1k-384px.py
Comment (fangyixiao18) [inline mmpretrain/configs/swin_transformer/swin-base_16xb64_in1k-384px.py:1]: please check all file names, and change all '-' to '_'

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]
please check all file names, and change all '-' to '_'

**Comment by timerring** [comment]
> please remove the folder `_base_/models/swin_transformer` and `_base_/models/swin_transformer_v2`, but only keep one file to set model config like `swin_transformer_base.py` and `swin_transformer_v2_base.py`,and the training configs all inherit from this `swin_transformer_base.py` file

Modifications have been completed as required @fangyixiao18

File: mmpretrain/configs/swin_transformer/swin_large_8xb8_cub_384px.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    from .._base_.models.swin_transformer_base import *
+    from .._base_.schedules.cub_bs64 import *
+
+# model settings
+checkpoint = 'https://download.openmmlab.com/mmclassification/v0/swin-transformer/convert/swin-large_3rdparty_in21k-384px.pth'  # noqa
+
+model = dict(
+    type=ImageClassifier,
+    backbone=dict(
+        init_cfg=dict(
```
Comment (fangyixiao18) [inline mmpretrain/configs/swin_transformer/swin_large_8xb8_cub_384px.py:22]: set arch

File: mmpretrain/configs/swin_transformer_v2/swinv2_large_w12_8xb128_in21k_192px.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+with read_base():
+    from .._base_.datasets.imagenet21k_bs128 import *
+    from .._base_.default_runtime import *
+    from .._base_.models.swin_transformer_v2_base import *
+    from .._base_.schedules.imagenet_bs1024_adamw_swin import *
+
+# model settings
+model = dict(
+    backbone=dict(img_size=192, window_size=[12, 12, 12, 6]),
```
Comment (fangyixiao18) [inline mmpretrain/configs/swin_transformer_v2/swinv2_large_w12_8xb128_in21k_192px.py:13]: set arch, please check other configs

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: mmpretrain/configs/swin_transformer_v2/swinv2_large_w12_8xb128_in21k_192px.py (hunk 2 above)
Comment (timerring) [reply mmpretrain/configs/swin_transformer_v2/swinv2_large_w12_8xb128_in21k_192px.py:13]: > set arch, please check other configs

The content here is consistent with [configs/swin_transformer_v2/swinv2-base-w12_8xb128_in21k-192px.py](https://github.com/open-mmlab/mmpretrain/blob/main/configs/swin_transformer_v2/swinv2-base-w12_8xb128_in21k-192px.py) before transplantation. In the original file, although `base_256.py` was introduced, `model` was redefined in the new file. In fact, the `model` content of `base_256.py` has been covered. I am wondering whether it is necessary to set arch here? @fangyixiao18

**Comment by timerring** [comment]
Modifications have been completed as required @fangyixiao18

**Review by fangyixiao18 — APPROVED** [review]

**Review by mzr1996 — APPROVED** [review]

**Pull request merged**